File: pset6/mario.py
from cs50 import get_int

def receberAltura():
  while True:
      altura = get_int("Digite a altura: ")

      if 0 < altura < 9:
        return altura
      else:
        print(f"Apenas altura entre 1 e 8.")

altura = receberAltura()
linha = 0

for i in range(altura):
  linha += 1
  print("  " * (altura - linha), end="")
  print("# " * linha, end="")
  print("# " * (linha - altura), end="")
  print("# " * linha, end="")
  print()

 

File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>


int main(void){

  int altura;
  int linha;
  int coluna;
  //Pede a altura ao usuário(valores entre 1 e 8)
  printf("===BEM VINDO AO JOGO DO MARIO!===\n");

  do {
    altura = get_int("Altura: ");
  } while (altura > 8 || altura < 1);

  //Loop da linhas
  for (linha = altura; linha > 0; linha--) {
    //Loop das colunas
    for (coluna = 0; coluna <= altura; coluna++) {
      if (coluna >= linha) {
        printf("#");
      } else if (coluna > 0) {
        printf(" ");
      }
    }
    printf("  ");
    for (coluna = altura; coluna >= 0; coluna--) {
      if (coluna >= linha) {
        printf("#");
      }
    }
    printf("\n");
  }
}
